jupyterlab_bigquery/jupyterlab_bigquery/details_handler/details_handler.py

```diff
@@ -54,100 +55,158 @@ def get_dataset_details(client, dataset_id):
+        'type': field.field_type,
+        'description': field.description,
+        'mode': field.mode
     })
 
+
 def format_detail_schema(schema):
   formatted_schema = []
   for field in schema:
     format_detail_field(formatted_schema, field, '')
   return formatted_schema
-    
+
 
 def get_table_details(client, table_id):
   table = client.get_table(table_id)
   return {
       'details': {
-          'id': "{}.{}.{}".format(table.project, table.dataset_id, table.table_id),
-          'name': table.table_id,
-          'description': table.description,
-          'labels': ["{}: {}".format(label, value) for label, value in table.labels.items()] if table.labels else None,
-          'date_created': json.dumps(table.created.strftime('%b %e, %G, %l:%M:%S %p'))[1:-1],
-          'expires': json.dumps(table.expires.strftime('%b %e, %G, %l:%M:%S %p'))[1:-1] if table.expires else None,
-          'location': table.location,
-          'last_modified': json.dumps(table.modified.strftime('%b %e, %G, %l:%M:%S %p'))[1:-1],
-          'project': table.project,
-          'dataset': table.dataset_id,
-          'link': table.self_link,
-          'num_rows': table.num_rows,
-          'num_bytes': table.num_bytes,
-          'schema': format_detail_schema(table.schema)
+          'id':
+              "{}.{}.{}".format(table.project, table.dataset_id,
+                                table.table_id),
+          'name':
+              table.table_id,
+          'description':
+              table.description,
+          'labels': [
+              "{}: {}".format(label, value)
+              for label, value in table.labels.items()
+          ] if table.labels else None,
+          'date_created':
+              json.dumps(table.created.strftime('%b %e, %G, %l:%M:%S %p'))
+              [1:-1],
+          'expires':
+              json.dumps(table.expires.strftime('%b %e, %G, %l:%M:%S %p'))[1:-1]
+              if table.expires else None,
+          'location':
+              table.location,
+          'last_modified':
+              json.dumps(table.modified.strftime('%b %e, %G, %l:%M:%S %p'))
+              [1:-1],
+          'project':
+              table.project,
+          'dataset':
+              table.dataset_id,
+          'link':
+              table.self_link,
+          'num_rows':
+              table.num_rows,
+          'num_bytes':
+              table.num_bytes,
+          'schema':
+              format_detail_schema(table.schema)
       }
   }
 
+
 def format_preview_field(formatted_fields, field, field_name):
   if field.field_type == 'RECORD':
-    for record_entry in field.fields:
-      format_preview_field(formatted_fields, record_entry, field_name + field.name + ".")
+    if field.mode == 'REPEATED':
+      formatted_fields.append(field.name)
+    else:
+      for record_entry in field.fields:
+        format_preview_field(formatted_fields, record_entry,
+                             field_name + field.name + ".")
```

Comment:
Repeated fields that are records won't get expanded out into multiple columns for each subfield anymore, since their repeated items will just be listed as a string.